File: 4.8-bc/py/stage-3-hexagonal/app/auth/domain/repository.py
"""Auth domain — UserRepository port (Protocol).

Implementations live in ``app.auth.infra.postgres``. Domain doesn't know
about pgx/asyncpg.
"""
from __future__ import annotations

from typing import Protocol
from uuid import UUID

from app.auth.domain.user import User


class UserRepository(Protocol):
    async def create(self, u: User) -> None: ...
    async def find_by_email(self, email: str) -> User | None: ...
    async def find_by_id(self, user_id: UUID) -> User | None: ...

File: 4.8-bc/ts/stage-3-hexagonal/src/auth/domain/repository.ts
// UserRepository — port. Реалізація живе в auth/infra/postgres.
import type { User } from "./user.js";

export interface UserRepository {
  create(u: User): Promise<void>;
  findByEmail(email: string): Promise<User | null>;
  findById(id: string): Promise<User | null>;
}
